tests/test_moe/test_moe_checkpoint.py

```diff
@@ -66,51 +136,83 @@ def get_model(parallel):
             custom_policy=OpenMoeForCausalLMPolicy(),
         )
     booster = Booster(plugin=plugin)
-    model, _, _, _, _ = booster.boost(model=model)
-    return model, booster
+    model, optim, _, _, _ = booster.boost(model=model, optimizer=optim)
+    return model, booster, optim
+
 
+def _test_moe_checkpoint(rank, parallel, shard):
+    if dist.get_rank() == 0:
+        if os.path.exists("./tmp_ckpt"):
+            shutil.rmtree("./tmp_ckpt")
+        if os.path.exists("./tmp_ckpt.pth"):
+            os.remove("./tmp_ckpt.pth")
+    dist.barrier()
 
-def _test_moe_checkpoint(parallel, shard):
     if parallel == None:
         MOE_MANAGER.setup(
-            seed=42,
             parallel=None,
         )
-    elif parallel == "zero2_ep":
+    elif parallel == "ep":
         MOE_MANAGER.setup(
-            seed=42,
             parallel="EP",
         )
+    elif parallel == "ep_zero":
+        MOE_MANAGER.setup(
+            parallel="EP",
+            max_ep_size=2,
+        )
     elif parallel == "hybrid":
         MOE_MANAGER.setup(
-            seed=42,
             parallel="EP",
             mode="fixed",
             fixed_dp_size=1,
             fixed_ep_size=2,
             fixed_pp_size=2,
         )
-    model1, booster1 = get_model(parallel)
-    model2, booster2 = get_model(parallel)
+    model1, booster1, optim1 = get_model(parallel)
```

Comment:
What's the difference of model1 and model2?

Reply:
Model1's optim will be updated and model2 will load model1's optim